### ISSUE TITLE
Failed to run plug&charge with EVerest software in the loop

### ISSUE THREAD

**Issue by SeanHsu0318** [EVerest/EVerest-archived#204]
I followed the instructions on https://everest.github.io/nightly/tutorials/how_to_plug_and_charge/index.html to try Plug & Charge with  EVerest software in the loop, but it seems some credential issues preventing it from working successfully.

When I run `bash ./run-scripts/run-sil-ocpp201-pnc.sh` or `bash ./run-scripts/run-sil-ocpp-pnc.sh`, I always find the error message "Failed to read cert_info! Not Accepted" in the CLI.

Has anyone successfully completed this simulation?

![image](https://github.com/user-attachments/assets/e7f9ee0a-9423-4954-9d8c-9f1b3d137f28)

**Comment by krealyt** [comment]
@Pietfried Did anybody already have such a scenario and eventually remember how to solve that?

**Comment by buzzik** [comment]
@SeanHsu0318 Hey man, have you solved this issue? I have the same

**Comment by Pietfried** [comment]
Hi @SeanHsu0318 @buzzik , this looks like you have not installed the required certificates or they cant be found by EVerest. Can you please check if there are certificates and keys installed inside your EVerest installation directory inside `dist/etc/everest/certs` ?

**Comment by buzzik** [comment]
@Pietfried nope in /ext/dist/share/everest/ I don't even have etc dir
But I have some certs in /ext/dist/etc/everest/certs

**Comment by Pietfried** [comment]
> [@Pietfried](https://github.com/Pietfried) nope in /ext/dist/share/everest/ I don't even have etc dir But I have some certs in /ext/dist/etc/everest/certs

Sorry, I made a mistake in my previous message and edited it. Your path looks correct. You can try to configure the paths to the respective certificates and keys manually using the module parameters of `EvseSecurity` individually. You can find the descriptions [here]( https://everest.github.io/nightly/_generated/modules/EvseSecurity.html)

The [more detailed description of the EvseSecurity module](https://everest.github.io/nightly/_included/modules_doc/EvseSecurity.html#everest-modules-handwritten-evsesecurity) might also help.

**Comment by buzzik** [comment]
@Pietfried can you help? I spent 2 days reading different docs and still can't launch it.
THese are certs I see in a docker
`/ext/dist/etc/everest/certs`
```sh
ls -R
.:
ca  client
./ca:
csms  cso  mf  mo  oem  v2g
./ca/csms:
CSMS_ROOT_CA.pem
./ca/cso:
./ca/mf:
MF_ROOT_CA.pem
./ca/mo:
MO_ROOT_CA.pem
./ca/oem:
./ca/v2g:
V2G_ROOT_CA.pem
./client:
cps  csms  cso  mf  mo  oem
./client/cps:
./client/csms:
./client/cso:
SECC_LEAF_M02_D13_Y2025_H11_M04_S05_i1_r960978141.key
./client/mf:
./client/mo:
./client/oem:
```

As I understood I need to have more certificates according to docs you provided, but... where to get them?

**Comment by Pietfried** [comment]
> [@Pietfried](https://github.com/Pietfried) can you help? I spent 2 days reading different docs and still can't launch it. THese are certs I see in a docker `/ext/dist/etc/everest/certs`
> 
> ls -R
> .:
> ca  client
> ./ca:
> csms  cso  mf  mo  oem  v2g
> ./ca/csms:
> CSMS_ROOT_CA.pem
> ./ca/cso:
> ./ca/mf:
> MF_ROOT_CA.pem
> ./ca/mo:
> MO_ROOT_CA.pem
> ./ca/oem:
> ./ca/v2g:
> V2G_ROOT_CA.pem
> ./client:
> cps  csms  cso  mf  mo  oem
> ./client/cps:
> ./client/csms:
> ./client/cso:
> SECC_LEAF_M02_D13_Y2025_H11_M04_S05_i1_r960978141.key
> ./client/mf:
> ./client/mo:
> ./client/oem:
> As I understood I need to have more certificates according to docs you provided, but... where to get them?

Yes, it looks like you are missing some certificates. I don't know how you installed everest-core, by default example certificates are generated based on the cmake option `ISO15118_2_GENERATE_AND_INSTALL_CERTIFICATES`.

You can generate the certificates using the following script: https://github.com/EVerest/ext-switchev-iso15118/blob/everest/iso15118/shared/pki/create_certs.sh

```bash
./create_certs.sh -v iso-2
```

You can then copy the generated `certs` folder into `/ext/dist/etc/everest/` . 

If you use https://github.com/EVerest/ocpp-csms for the CSMS, make sure to point to the same directory when starting the CSMS.

Hope this helps!
